### PR TITLE
[bhavanatumma/register-component-change] - register and deploy model …

### PR DESCRIPTION
**Pull request by BhavanaTumma**
…changes

### CODE REVIEW

File: assets/common/src/deploy.py [hunk 1]
```diff
@@ -260,9 +261,9 @@ def main():
     args = parse_args()
     ml_client = get_mlclient()
     # get registered model id
-    if args.registration_details:
+    if args.model_registration_details:
         model_info = {}
-        with open(args.registration_details) as f:
+        with open(args.model_registration_details) as f:
```
Comment (HrishikeshGeedMS) [inline assets/common/src/deploy.py:266]: It should be **os.path.join(model_registration_details,"model_registration_details")** no??

File: assets/common/components/deploy_model/spec.yaml
Comment (HrishikeshGeedMS) [inline assets/common/components/deploy_model/spec.yaml:1]: Can we change output of deployment component to uri_folder in same way as we are doing for registration.
In future we may face same issue with deployment component.

File: assets/common/components/register_model/spec.yaml
Comment (HrishikeshGeedMS) [inline assets/common/components/register_model/spec.yaml:1]: version needs to be changed for this component. Also make sure to add new version in import model as well

File: assets/common/components/deploy_model/spec.yaml
Comment (HrishikeshGeedMS) [reply assets/common/components/deploy_model/spec.yaml:1]: Change version for this component

**Review by HrishikeshGeedMS — CHANGES_REQUESTED** [review]

**Review by HrishikeshGeedMS — APPROVED** [review]

File: assets/common/components/register_model/spec.yaml [hunk 2]
```diff
@@ -77,8 +77,8 @@ inputs:
 
 outputs:
   registration_details:
-    type: uri_file
-    description: JSON file into which model registration details will be written
+    type: uri_folder
+    description: A folder which contains a JSON file into which model registration details will be written
```
Comment (novaturient95) [inline assets/common/components/register_model/spec.yaml:81]:
```suggestion
    description: Folder containing model registration details in a JSON file
```

File: assets/common/components/register_model/spec.yaml [hunk 3]
```diff
@@ -77,8 +77,8 @@ inputs:
 
 outputs:
   registration_details:
```
Comment (novaturient95) [inline assets/common/components/register_model/spec.yaml:79]:
```suggestion
  registration_details_folder:
```

File: assets/common/components/deploy_model/spec.yaml [hunk 4]
```diff
@@ -39,9 +39,9 @@ command: >-
 inputs:
   # Output of registering component
   registration_details:
```
Comment (novaturient95) [inline assets/common/components/deploy_model/spec.yaml:41]:
```suggestion
  registration_details_folder:
```

File: assets/common/components/deploy_model/spec.yaml [hunk 5]
```diff
@@ -39,9 +39,9 @@ command: >-
...
-    type: uri_file
+    type: uri_folder
     optional: true
-    description: Json file that contains the ID of registered model to be deployed
+    description: A folder that contains a Json file that has the ID of registered model to be deployed
```
Comment (novaturient95) [inline assets/common/components/deploy_model/spec.yaml:44]:
```suggestion
    description: Folder containing model registration details in a JSON file
```

File: assets/common/src/deploy.py [hunk 6]
```diff
@@ -48,7 +49,7 @@ def parse_args():
     parser.add_argument(
         "--registration_details",
         type=Path,
-        help="Json file that contains the ID of registered model to be deployed",
+        help="A folder thant contains a Json file that has the ID of registered model to be deployed",
```
Comment (novaturient95) [inline assets/common/src/deploy.py:52]:
```suggestion
        help="Folder containing model registration details in a JSON file",
```

File: assets/common/src/deploy.py [hunk 7]
```diff
@@ -262,10 +263,15 @@ def main():
     # get registered model id
     if args.registration_details:
         model_info = {}
-        with open(args.registration_details) as f:
-            model_info = json.load(f)
-        model_id = model_info["id"]
-        model_name = model_info["name"]
+
+        registration_details_file = args.registration_details/"model_registration_details.json"
```
Comment (novaturient95) [inline assets/common/src/deploy.py:267]:
```suggestion
        registration_details_file = args.registration_details / "model_registration_details.json"
```

Comment (novaturient95) [inline assets/common/src/deploy.py:267]: Declare `model_registration_details.json` in constants and use it in both register and deploy model launcher scripts

File: assets/common/src/deploy.py [hunk 8]
```diff
@@ -262,10 +263,15 @@ def main():
...
+        try:
```
Comment (novaturient95) [inline assets/common/src/deploy.py:268]: Could you verify for following cases:
1. registration_details_folder  exists, but `model_registration_details.json` doesn't and `model_id` is passed
2. registration_details_folder  exists, but `model_registration_details.json` doesn't and `model_id` is not passed
3. registration_details_folder  and `model_registration_details.json` exists

File: assets/common/src/register.py [hunk 9]
```diff
@@ -211,7 +212,9 @@ def main():
     }
     json_object = json.dumps(model_info, indent=4)
 
-    with open(registration_details, "w") as outfile:
+    registration_file =  os.path.join(registration_details, "model_registration_details.json")
```
Comment (novaturient95) [inline assets/common/src/register.py:215]: Please declare `model_registration_details.json` in constants.
nit: For `Path` variable use `/` op for joining over `os.path.join`

File: assets/common/src/register.py [hunk 10]
```diff
@@ -211,7 +212,9 @@ def main():
...
+
+    with open(registration_file, "w") as outfile:
```
Comment (novaturient95) [inline assets/common/src/register.py:217]: As discussed, please add this in try-except block to make sure component does not fail for any issue at this point

**Review by novaturient95 — CHANGES_REQUESTED** [review]

File: assets/common/src/register.py (hunk 10 above)
Comment (BhavanaTumma) [reply assets/common/src/register.py]: Done

File: assets/common/src/deploy.py (hunk 8 above)
Comment (BhavanaTumma) [reply assets/common/src/deploy.py:268]: Added.

File: assets/common/src/deploy.py (hunk 6 above)
Comment (BhavanaTumma) [reply assets/common/src/deploy.py]: Done.

File: assets/common/components/register_model/spec.yaml
Comment (BhavanaTumma) [reply assets/common/components/register_model/spec.yaml:1]: Done

File: assets/common/components/deploy_model/spec.yaml
Comment (BhavanaTumma) [reply assets/common/components/deploy_model/spec.yaml:1]: We'll change the deploy model output at a later point. version change has been done.

File: assets/common/src/deploy.py [hunk 11]
```diff
@@ -260,12 +261,17 @@ def main():
     args = parse_args()
     ml_client = get_mlclient()
     # get registered model id
-    if args.registration_details:
+    if args.registration_details_folder and \
+            os.path.exists(args.registration_details_folder/ComponentVariables.REGISTRATION_DETAILS_JSON_FILE):
```
Comment (novaturient95) [inline assets/common/src/deploy.py:265]: Please use something like below for readibility: 

```
registration_details_file = args.registration_details_folder / ComponentVariables.REGISTRATION_DETAILS_JSON_FILE
if registration_details_file.exists():
   ....

Comment (BhavanaTumma) [reply assets/common/src/deploy.py]: Done

**Review by HrishikeshGeedMS — APPROVED** [review]

**Review by elesel — APPROVED** [review]

**Review by novaturient95 — APPROVED** [review]

**Pull request merged**